lib/valueflow.cpp

```diff
@@ -917,9 +917,9 @@ static void setTokenValue(Token* tok,
                 const double floatValue1 = value1.isFloatValue() ? value1.floatValue : value1.intvalue;
                 const double floatValue2 = value2.isFloatValue() ? value2.floatValue : value2.intvalue;
                 const MathLib::bigint intValue1 =
-                    value1.isFloatValue() ? std::llround(value1.floatValue) : value1.intvalue;
```

Comment:
Maybe mention that UBSAN warning in comment that it is intentional.

Reply:
I doubt that the ubsan warning is intentional. if `value1.floatValue` is too large and we try to cast it to bigint.. what is the result? we don't want a random value for it right?

Reply:
> if value1.floatValue is too large and we try to cast it to bigint.. what is the result? we don't want a random value for it right?

We dont actually use the value since one of the values is already float. I think we could make this lazy.